lib/elixir/lib/kernel.ex

```diff
@@ -942,6 +943,13 @@ defmodule Kernel do
       iex> 'foo' ++ 'bar'
       'foobar'
 
+      # returns an improper list
+      iex> [1] ++ 2
+      [1 | 2]
+
+      # returns a proper list
+      iex> [1] ++ [2]
+      [1, 2]
```

Comment:
well, yes, but i just want to show how it should be written to return a proper list for the same case as when an improper list is returned